1_fetch/src/get_sb_data.R

```diff
@@ -1,177 +1,42 @@
-get_babies <- function(idin) {
-  listall <- item_list_children(idin, limit = 999)
-  listall2 <- lapply(listall, `[`, c('id'))
-  ids <- unlist(listall2)
-  return(ids)
-}
 
-make_dl_table <- function(sb_parent_id, outdir) {
-  
-  #start with doing the top level id so we have something to work with.
-  idlist <- get_babies(sb_parent_id)
-  
-  #these are our 2 working lists for our loop
-  #the = "blank" thing is maybe a little lazy, but i take care of it later.
-  parlist <- idlist
-  dllist <- "blank"
-  
-  #getting ready for our loop
-  end <- FALSE
-  count <- 1
-  
-  #the main stuff
-  #the first if just ends the loop when you go past the end of the parlist
-  #otherwise, we test the id in parlist. We use item_get_fields to check if the id has children.
-  #if yes, we get the ids of those children with our get_babies function
-  #and we add them to the end of parlist so we can check those ids later in our loop
-  #if the id has no children, that means its an id with download links.
-  #and we add them to our dllist.
-  while (end == FALSE)
-  {
-    if (is.na(parlist[count]) == TRUE)
-    {
-      end <- TRUE
-      print("we are done")
-    }
-    if (is.na(parlist[count]) == FALSE)
-    {
-      idN <- parlist[count]
-      print(paste("trying", idN))
-      test <- item_get_fields(idN,'hasChildren')
-      if (test == TRUE)
-      {
-        templist <- get_babies(idN)
-        parlist <- c(parlist, templist)
-        print("adding to parlist")
-      }
-      if (test == FALSE)
-      {
-        print("adding one to dllist")
-        dllist <- c(dllist, idN)
-      }
-      count <- count + 1
-    }
-  }
-  
-  #this gets rid of our first "blank" entry
-  dllist <- dllist[-1]
+get_sb_data <- function(sites, sb_var_ids, dldir, workdir, outdir, out_file_name) {
```

Comment:
I was thinking of that for our other log files as well - nothing depends on them, so it would be possible to miss changes to the log files. I think it should be a separate issue.

Reply:
Okay, sounds good. Part of issue #60 included this download check. I'll add a comment on that issue that we're moving the check to a new issue